common/src/ethereum/walletconnect.rs

```diff
@@ -0,0 +1,104 @@
+use std::error::Error;
+use walletconnect::client::{CallError, SessionError};
+use walletconnect::{qr, Client, Metadata, Transaction};
+
+use ethers::types::transaction::eip2718::TypedTransaction;
+use ethers::types::{Address, NameOrAddress, Signature, U256};
+
+#[derive(Debug)]
+pub struct WalletConnect {
+    client: Client,
+    chain_id: u64,
+    address: Address,
+}
+
+impl WalletConnect {
+    pub fn new() -> Result<Self, Box<dyn Error>> {
+        let client = Client::new(
+            "rad-ens",
```

Comment:
What's this? :thinking: Profile? Any idea what this is supposed to be?

Reply:
This is the profile that's shown when walletconnect wants permission to connect with your app.

![Screenshot from 2022-03-28 12-21-29](https://user-images.githubusercontent.com/14107758/160351737-81c19e83-3e60-463d-922c-44504059a76b.png)

Reply:
But where does the string `rad-ens` show in this case?

Reply:
If we don't know, maybe set it to `radicle-cli` :shrug:

Reply:
https://github.com/radicle-dev/radicle-cli/commit/ce9ad2d97fec16a5ef720cbb82ae87a96f6f5e28#diff-363b2cf3885221c5b26bfa5862fc3e2363454ef177a777c5e11759468b8edc5cR18